src/rate_limiter/src/lib.rs

```diff
@@ -132,7 +132,8 @@ impl TokenBucket {
         // refill_token_count = (delta_time * size) / (complete_refill_time_ms * 1_000_000)
         // In order to avoid overflows, simplify the fractions by computing greatest common divisor.
 
-        let complete_refill_time_ns = complete_refill_time_ms * NANOSEC_IN_ONE_MILLISEC;
+        let complete_refill_time_ns =
```

Comment:
nitpick: add a unit test case for early error thrown here.

Reply:
As synced offline, the testing for this will be provided in a follow up